### DISCUSSION TITLE
How do you integrate Laravel Tags with laravel-sluggable?

### DISCUSSION THREAD

**Discussion by Rocketpilot** [spatie/laravel-tags#516] [Q&A]
I've been having trouble getting routing via slug working with Laravel Tags. I haven't created a custom model, which would probably work, but the docs don't give a complete example and I'm not an advanced enough developer to fill in the missing pieces. 

I see there's a config setting for choosing which slugger to use, but again the documentation stops before it spells out how to do it. 

Great package, when it works, it works flawlessly but there are edge cases that really stick out.

**Comment by michaelbeers** [comment]
Just create an custom `Tag` model and extend `Spatie\Tags\Tag` after that you can override the `getRouteKeyName()` function

```php
<?php
namespace App\Models;

use Spatie\Tags\Tag as BaseTag;

class Tag extends BaseTag
{
    /**
     * Get the route key for the model.
     */
    public function getRouteKeyName(): string
    {
        return 'slug';
    }
}
```

Finally change the `tag_model` config to `App\Models\Tag::class`, that should to the job.

All code is untested :)
